src/devcenter/azext_devcenter/custom.py

```diff
@@ -186,6 +211,14 @@ def _cli_arguments_loader(self):
 
 
 class DevBoxDefinitionCreate(_DevBoxDefinitionCreate):
+    @classmethod
+    def _build_arguments_schema(cls, *args, **kwargs):
+        args_schema = super()._build_arguments_schema(*args, **kwargs)
+        args_schema.image_reference._required = True
+        args_schema.sku._required = True
+        args_schema.os_storage_type._required = True
```

Comment:
Why not make these properties required in swagger?

Reply:
These whole body object is marked as required in the swagger (https://github.com/Azure/azure-rest-api-specs/blob/97a3d07807c5e7f23e0f47f532d0555ff2aa5d5d/specification/devcenter/resource-manager/Microsoft.DevCenter/stable/2023-04-01/devcenter.json#L2347-L2357)

But when using aaz, the required property did not transfer to all the properties. Can add this as TODO to investigate on our side, but can't change the current swagger in time for our release date. Same with the other similar comments

Reply:
The `properties` is not required that's why didn't transfer to its sub properties.

Reply:
You can fix it next time.